src/Symfony/Component/TypeInfo/BuiltinType.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\TypeInfo;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ * @author Baptiste Leduc <baptiste.leduc@gmail.com>
+ */
+enum BuiltinType: string
```

Comment:
```suggestion
enum BuiltinTypeEnum: string
```

Reply:
I see no point justifying this, I would prefer the enum to stay like it is.

Reply:
I see one: DX as there are 2 classes named `BuiltinType` currently which is painful when looking for one via IDE' search feature. Also you're aliasing this to `BuiltinTypeEnum` in `BuiltinType` (the other, here we are :)) which makes that part of the patch harder to follow.

Reply:
Thx for the use case examples. I now think it's oke to keep it without the suffix.

Looking at https://www.php.net/manual/en/language.types.type-system.php#language.types.type-system.atomic
The enum includes more then the built-in types maybe another option is to name it `BaseType`

```php
Type::builtin(BaseType::INT);
```

Reply:
Fair point, we do have either built-in, value, and aliases.

I'm not really for `BaseXXX` because it makes me think about inheritance.

PHPStan seems to document it as [basic types](https://phpstan.org/writing-php-code/phpdoc-types#basic-types), and to name it `IdentifierType` in the codebase.

`TypeIdentifier` is a good candidate for me. Indeed, as it doesn't end with "Type", it cannot be confused with other types.

Reply:
`TypeName` could be a great candidate too